simfection/interaction_engine.py

```diff
@@ -152,24 +161,92 @@ def _interact(self, a, b):
+            pop_days_infected = self.population['days_infected'].tolist()
+            pop_immunity = self.population['immunity'].tolist()
+
+            # Set the Population in the PyInteractions object
+            interactions.set_population(pop_agent,
+                                        pop_state,
+                                        pop_infected_by,
+                                        pop_days_infected,
+                                        pop_immunity)
+
+            # Get all the connections information required for running PyInteractions
+            conn_agent = self.connections['agent'].tolist()
+            conn_connections = self.connections['connections'].tolist()
+            conn_num_connections = self.connections['num_connections'].tolist()
+            conn_max_connections = self.connections['max_connections'].tolist()
+
+            # Set the Connections in the PyInteractions object
+            interactions.set_connections(conn_agent,
+                                         conn_connections,
+                                         conn_num_connections,
+                                         conn_max_connections)
+
+            # Set the Pathogen settings, first convert each value in pathogen
+            # to unicode bytes for ingestion to interactions
+            pathogen = {key.encode(): value for key, value in self.pathogen.items()}
+            interactions.set_pathogen_settings(pathogen)
+
+            # Run the Interaction Engine functionality
+            interactions.interact_all()
+            results = interactions.get_population_df()
+            self.population = results
+            if verbose:
+                logger.debug(
+                    '- Population states: {}'
+                    .format(
+                        self
+                        .population
+                        .state
+                        .value_counts(normalize=True)
+                        .to_dict()
+                    )
+                )
+        else:
+            logger.debug('- Getting unique interactions.')
+            self._get_unique_connections()
+            connections = self.connections
+            logger.debug('- Running all unique interactions.')
+            for a, b in connections.interaction.values:
+                self._interact(a, b)
+
+            if verbose:
+                logger.debug(
+                    '- Population states: {}'
+                    .format(
+                        self
+                        .population
+                        .state
+                        .value_counts(normalize=True)
+                        .to_dict()
+                    )
+                )
 
 
 pathogen = {
     'infection_rate': 0.4,
     'recovery_rate': 0.1,
     'death_rate': 0.02,
     'spontaneous_rate': 0.0,
-    'testing_accuracy': None,
+    'testing_accuracy': 0,
```

Comment:
Is there a reasons this was changed?

Reply:
There's nothing in the codebase that uses it, and None doesn't exist in C++ land. I've just updated the API to reset any None types to -1 for error handling in C++.